dlt/pipeline/pipeline.py

```diff
@@ -1765,14 +1765,17 @@ def dataset(
             logger.info(
                 f"Make sure that tables declared in explicit schema {schema.name} are present on"
                 f" dataset {self.dataset_name}"
-            )
-        elif not self.default_schema_name:
-            raise PipelineNeverRan(self.pipeline_name, self.pipelines_dir)
-        elif schema is None:
-            schema = self.default_schema
+            )  #
         elif isinstance(schema, str):
-            # schema with given name must be present
-            schema = self.schemas[schema]
+            if schema := self.schemas.get(schema, None):
+                pass
+            else:
```

Comment:
I'd like to allow to get a dataset with a schema not known by the pipeline yet, because then I can use it to see wether it is new or on the same destination before it was ever written. This makes a few tests fail which would be easy to fix but i'm not sure if I should maybe add one more param that you have to set to allow this?